flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/planner/delegation/hive/HiveParserDMLHelper.java

```diff
@@ -331,20 +333,48 @@ public Operation createInsertOperation(HiveParserCalcitePlanner analyzer, RelNod
                 createInsertOperationInfo(
                         queryRelNode,
                         destTable,
-                        HiveParserBaseSemanticAnalyzer.parseCompoundName(catalogManager, tableName),
+                        HiveParserBaseSemanticAnalyzer.parseCompoundName(
+                                catalogRegistry, tableName),
                         staticPartSpec,
                         analyzer.getDestSchemaForClause(insClauseName),
                         overwrite);
 
         return new SinkModifyOperation(
-                catalogManager.getTableOrError(insertOperationInfo.f0),
+                getContextResolvedTable(insertOperationInfo.f0),
                 insertOperationInfo.f1,
                 insertOperationInfo.f2,
                 null, // targetColumns
                 insertOperationInfo.f3,
                 Collections.emptyMap());
     }
 
+    private ContextResolvedTable getContextResolvedTable(ObjectIdentifier tableIdentifier) {
```

Comment:
Yes, we still needs to depend on `ContextResolvedTable` for it's need to construct `SinkModifyOperation` which is also `@Internal`. In fact, Hive dialect still use many internal operations. Maybe we need to expose them to public then,  but it may be another topic. 
Considering our plan is decoupling Hive with flink-table-planner in the first, I think we can keep them to see how much these internal  interfaces/classes will affect the maintainance/development of Hive dialect to take further steps.
Btw, it make me be aware of the title of the pr/jira is not correct,  it should be `Hive dialect shouldn't use interfaces in table-planner to build LogicPlan`.